### DISCUSSION TITLE
Post request with EntityView as parameter using Quarkus

### DISCUSSION THREAD

**Discussion by chrigiansen** [Blazebit/blaze-persistence#1613] [Q&A]
Hello

I am currently trying to get my application to update / create entities using an EntityView as parameter.

Unfortunately, I am getting this Exception:
`
com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of 'ch.chrigiansen.quarkus.blazebit.rest.view.GiftUpdateView' (no Creators, like default constructor, exist): abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information
 at [Source: (io.quarkus.vertx.http.runtime.VertxInputStream); line: 1, column: 1]
`

This brought me to the attention of this [Quarkus Guide](https://quarkus.io/guides/rest-json). I did implement such a CustomObjectMapper class, but it is not used as intended. 

My implementation looks like this:
``` java
public class BlazeObjectMapper {

	@Inject
	EntityViewManager entityViewManager;

	@Produces
	ObjectMapper objectMapper(Instance<ObjectMapperCustomizer> customizers) {
		ObjectMapper mapper = new EntityViewAwareObjectMapper(entityViewManager, new ObjectMapper()).getObjectMapper();

		customizers.forEach(c -> c.customize(mapper));

		return mapper;
	}
}
```
I am having two issues:
- Well, it seems to not have any effect
- In the original Quarkus guide, the method should be a singleton, but that creates another issue.

I guess the issue in my example is, that the returned mapper is only a subset of the EntityViewAwareObjectMapper and therefore not capable of providing a new instance for the json. 

I will provide some [code](https://github.com/Blazebit/blaze-persistence/files/9924610/quarkus.blazebit.rest.zip) to you, which you can run with maven. There are two tests, one for a get that works (for good vibes) and the post test that fails. Could you be so kind as to fix what is missing and provide the code back? Besides actually needing this functionality, I plan on enriching the guide on the Quarkus page and provide code to the repository. 

Many thanks in advance

Chrigiansen

**Comment by beikov** [comment]
Hi there. I didn't look into the code yet, but can you try setting up the JAXRS-Jackson integration please? https://persistence.blazebit.com/documentation/1.6/entity-view/manual/en_US/#jaxrs-setup

You shouldn't have to implement custom object mappers. Also take a look into the [sample app](https://github.com/Blazebit/blaze-persistence/tree/main/examples/quarkus/base). Maybe that helps you figure out what is missing already.

Please let us all know what your problem was so I can improve the documentation.

**Reply by chrigiansen** [reply]
Thank you very much. It works now. 

I did forget the `blaze-persistence-integration-jaxrs-jackson` dependency. What did confuse me was, that I got the rest interface to work with blazebit for get and therefore ruled out an issue with jaxrs. I was therefore only looking into chapter 18.1 Jackson Integration. Maybe having more overarching examples would help. i.e. Combine the jax-rs, jackson and hibernate dependency into one.

Something like this.

``` xml
		<dependency>
			<groupId>com.blazebit</groupId>
			<artifactId>blaze-persistence-integration-quarkus</artifactId>
		</dependency>

		<dependency>
			<groupId>com.blazebit</groupId>
			<artifactId>blaze-persistence-integration-jaxrs</artifactId>
			<version>${blaze-persistence.version}</version>
		</dependency>

		<dependency>
			<groupId>com.blazebit</groupId>
			<artifactId>blaze-persistence-integration-jackson</artifactId>
			<version>${blaze-persistence.version}</version>
		</dependency>

		<dependency>
			<groupId>com.blazebit</groupId>
			<artifactId>blaze-persistence-integration-jaxrs-jackson</artifactId>
			<version>${blaze-persistence.version}</version>
			<scope>runtime</scope>
		</dependency>

		<dependency>
			<groupId>com.blazebit</groupId>
			<artifactId>blaze-persistence-integration-hibernate-5.6</artifactId>
			<scope>runtime</scope>
		</dependency>
```

Shorty I will use the working code to enhance the Quarkus guide regarding Blazebit. Currently there is no code and the snippets don't compile.

Thanks Again

**Reply by beikov** [reply]
Well, I understand that it is a bit unfortunate that one needs to consider so many different integrations, but the pluggability in Quarkus etc. kind of makes this necessary 😅
